[System.Data] SqlTypes and DataStorage from corefx

EgorBo opened the pull request:
Use corefx sources when it is possible in System.Data (SqlTypes, DataStorage and its subclasses)

dnfclas commented:
Hi __@EgorBo__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

dnfclas commented:
__@EgorBo__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

akoeplinger commented on mcs/class/System.Data/ReferenceSources/SR.cs:
```diff
@@ -0,0 +1,1429 @@
+	static partial class SR
```
We have a tool in mcs/tools/resx2sr that you can use to turn the Strings.resx from corefx into SR.cs, it outputs a simpler format than this one.

EgorBo replied:
@akoeplinger good to know, thanks! Res.cs confused me

marek-safar replied:
Use resx2sr and put generated file to corefx/SR.cs instead

akoeplinger replied:
@EgorBo this file should be deleted, it is unused now.

akoeplinger commented on mcs/class/System.Data/System.Data-net_4_x.csproj:
```diff
@@ -49,6 +49,26 @@
   </PropertyGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <ItemGroup>
+    <Compile Include="..\..\..\external\corefx\src\System.Data.Common\src\System\Data\SQLTypes\INullable.cs" />
```
As mentioned on the top of the file, the .csproj's are autogenerated so you don't really need to modify it manually. We have a bot that updates them on demand (the csproj's are not used by our build).

EgorBo replied:
@akoeplinger I didn't as far as I remember, 'make update-solution-files' did it :)

akoeplinger replied:
@EgorBo ah ok, yeah that's what the bot does as well :) I usually don't run it out of convenience and let the bot handle it after the PR is merged.

akoeplinger commented on mcs/class/System.Data/mobile_referencesource.sources:
```diff
@@ -270,23 +270,23 @@ MobileStubs.cs
+../../../external/corefx/src/System.Data.Common/src/System/Data/SQLTypes/SQLSingle.cs
+../../../external/corefx/src/System.Data.Common/src/System/Data/SQLTypes/SQLString.cs
+../../../external/corefx/src/System.Data.Common/src/System/Data/SQLTypes/SQLUtility.cs
+../../../external/corefx/src/System.Data.Common/src/System/Data/SQLTypes/SqlXml.cs
```
Please make sure .sources files end with newlines (it screws up some part of our build on Linux otherwise).

akoeplinger commented on mcs/class/System.Data/referencesource.sources:
```diff
@@ -18,19 +18,12 @@
 ../referencesource/System.Data/System/Data/Common/ActivityCorrelator.cs
```
Doesn't need to be part of this PR, but I think it'd be good to extract the files which are duplicated between referencesource.sources and mobile_referencesource.sources into a common.sources file.

EgorBo replied:
@akoeplinger can I use corefx.sources name instead?

akoeplinger replied:
@EgorBo Sure.

EgorBo commented:
@akoeplinger @marek-safar I've updated my PR.
1) Generated SR.cs using resx2sr tool and put it to corefx/SR.cs
2) Fixed line endings
3) mobile_referencesource.sources used to 99% duplicate referencesource.sources. I've removed both and moved the common part to corefx.common.sources file.

EgorBo commented:
@akoeplinger 
I wanted to delete the whole folder 'ReferenceSources' once I finish migration to corefx, but ok, I've just removed that file.

marek-safar commented on mcs/class/System.Data/mobile_System.Data.dll.sources:
```diff
@@ -64,6 +65,6 @@ System.Data.Common/DbTypes.cs
 
-#include mobile_referencesource.sources
-
+MobileStubs.cs
```
Where is this file? Also it should go to corefx/ folder if it's for corefx specific implementation

akoeplinger replied:
It was in mobile_referencesource.sources before, and is not corefx specific: https://github.com/mono/mono/blob/master/mcs/class/System.Data/MobileStubs.cs
